jOOQ/src/main/java/org/jooq/tools/jdbc/JDBCUtils.java

```diff
@@ -119,6 +129,66 @@ public static final SQLDialect dialect(Connection connection) {
         return result;
     }
 
+    public static final SQLDialect dialect(String url, int majorVersion, int minorVersion) {
+        SQLDialect dialect = dialect( url );
+
+        switch ( dialect ) {
+            case POSTGRES:
+                return getPostgreSqlDialect( majorVersion, minorVersion );
+            case MYSQL:
+                return getMySqlDialect( majorVersion, minorVersion );
+            case FIREBIRD:
+                return getFirebirdDialect( majorVersion, minorVersion );
+        }
+
+        return dialect;
+    }
+
+    public static final SQLDialect getPostgreSqlDialect(int majorVersion, int minorVersion) {
+        switch ( majorVersion ) {
+            case 10:
+                return POSTGRES_10;
+            case 9:
+                if ( minorVersion >= 5 ) {
+                    return POSTGRES_9_5;
+                }
+                else if ( minorVersion == 4 ) {
+                    return POSTGRES_9_4;
+                }
+                else if ( minorVersion == 3 ) {
+                    return POSTGRES_9_3;
+                }
```

Comment:
Hmm, another thing is that we probably want to return `POSTGRES_9_3` on all versions <= 9.3 as well, because `POSTGRES` (the dialect family) behaves like the latest version currently supported by jOOQ

Reply:
The same thing we should do for ```FIREBIRD``` doesn't it?

Reply:
Sure, I already did: https://github.com/jOOQ/jOOQ/commit/236c65b26db95eeb4c8eaa218ba201991dc9ba7c